telegram/ext/_handlers/chatmemberhandler.py

```diff
@@ -99,12 +113,31 @@ def check_update(self, update: object) -> bool:
             :obj:`bool`
 
         """
-        if isinstance(update, Update):
-            if not (update.my_chat_member or update.chat_member):
-                return False
-            if self.chat_member_types == self.ANY_CHAT_MEMBER:
-                return True
-            if self.chat_member_types == self.CHAT_MEMBER:
-                return bool(update.chat_member)
-            return bool(update.my_chat_member)
-        return False
+        if not isinstance(update, Update):
+            return False
+
+        if not (update.my_chat_member or update.chat_member):
+            return False
+
+        if self.__is_chat_restricted(update):
+            return False
+
+        if self.chat_member_types == self.ANY_CHAT_MEMBER:
+            return True
+
+        if self.chat_member_types == self.CHAT_MEMBER:
+            return bool(update.chat_member)
+
+        return bool(update.my_chat_member)
+
+    def __is_chat_restricted(self, update: Update) -> bool:
+        """Checks if the handler is chat ID restricted and doesn't match with update's chat ID."""
```

Comment:
I personally find the negation a bit confusing. I would find it more intuitive to have a function `__is_chat_allowed` …

Reply:
Had to introduce this function only because pylint was failing with so many `if` conditions in `check_update`. Your suggestion to use `update.effective_chat` instead resolved it. So moved the condition back to `check_update` and removed this function altogether. Fixed the other issues as well. Let me know if everything looks okay now :)